nntrainer/src/neuralnet.cpp

```diff
@@ -583,9 +583,23 @@ void NeuralNetwork::readModel() {
 
 int NeuralNetwork::train() {
   int status = ML_ERROR_NONE;
-  data_buffer->run(nntrainer::BUF_TRAIN);
-  data_buffer->run(nntrainer::BUF_VAL);
-  data_buffer->run(nntrainer::BUF_TEST);
+  status = data_buffer->run(nntrainer::BUF_TRAIN);
+  if (status) {
```

Comment:
`status != ML_ERROR_NONE` might be better?

Reply:
fixed.